pipeline/trusted-resources/cmd/sign/main.go

```diff
@@ -71,36 +73,49 @@ func main() {
 	}
 	defer f.Close()
 
-	tsBuf, err := ioutil.ReadFile(*taskFile)
+	tsBuf, err := ioutil.ReadFile(*resourceFile)
 	if err != nil {
 		log.Fatalf("error reading task: %v", err)
 	}
 
-	ts := &v1beta1.Task{}
-	if err := yaml.Unmarshal(tsBuf, &ts); err != nil {
+	var crd resourcesemantics.GenericCRD
+	switch *kind {
+	case "Task":
+		crd = &v1beta1.Task{}
+	case "Pipeline":
+		crd = &v1beta1.Pipeline{}
+	}
+	if err := yaml.Unmarshal(tsBuf, &crd); err != nil {
 		log.Fatalf("error unmarshalling taskrun: %v", err)
```

Comment:
Is the error message correct? should be something like`error unmarshalling Task/Pipeline`?

Reply:
ah thanks!